specification/security/query_roles/QueryRolesRequest.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { RoleAggregationContainer } from './types'
+import { RequestBase } from '@_types/Base'
+import { integer } from '@_types/Numeric'
+import { Sort, SortResults } from '@_types/sort'
+
+/**
+ * Retrieves roles in a paginated manner. You can optionally filter the results with a query.
+ * @rest_spec_name security.query_role
```

Comment:
Note this is singular here `query_role`, but the specs dir is plural (`query_roles`).
ES endpoints are singular, eg https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-query-api-key.html , but I see it in the specs that the dir is named `query_api_keys`.